src/object-pool.ts

```diff
@@ -1,10 +1,287 @@
+			this.parseOptionsState(options.state);
+		}
+
+		if (options.startSize !== undefined) {
+			this.parseOptionsStartSize(options.startSize);
+		}
+	}
+	public parseOptionsStartSize(startSize: number): void {
+		if (!this.isInteger(startSize)) {
+			return;
+		}
+		if (startSize < 0) {
+			return;
+		}
+
+		this.startSize = startSize;
+	}
+	public parseOptionsState(state: ArmorObjectPoolState<T> | string): void {
+		if (!state) {
+			return;
+		}
+
+		let result: ArmorObjectPoolState<T> | null = null;
+
+		if (typeof state === 'string') {
+			result = this.parse(state)!;
+
+			if (!result) {
+				throw new Error('state is not valid');
+			}
+		} else {
+			result = state;
+
+			if (state.type !== 'opState') {
+				throw new Error('state must be an ArmorObjectPoolState');
+			}
+			if (typeof result.autoIncrease !== 'boolean') {
+				throw new Error('state autoIncrease must be a boolean');
+			}
+
+			if (!this.isInteger(result.objectCount)) {
+				throw new Error('state objectCount must be an integer');
+			}
+			if (!this.isInteger(result.maxSize)) {
+				throw new Error('state maxSize must be an integer');
+			}
+
+			if (typeof result.increaseFactor !== 'number' || result.increaseFactor < 0) {
+				throw new Error('state increaseFactor must be a positive number');
+			}
+
+			const between0and1 = state.increaseBreakPoint >= 0 && state.increaseBreakPoint <= 1;
+			if (typeof result.increaseBreakPoint !== 'number' || !between0and1) {
+				throw new Error('state increaseBreakPoint must be a number between 0 and 1');
+			}
+		}
+
+		this.state.autoIncrease = result.autoIncrease;
+		this.state.maxSize = result.maxSize;
+		this.state.increaseFactor = result.increaseFactor;
+		this.state.increaseBreakPoint = result.increaseBreakPoint;
+
+		this.startSize = result.objectCount;
+	}
+
+	public utilization(n: number = 0): number {
+		if (!this.isValidState(this.state)) {
+			return NaN;
+		}
+		if (this.state.objectCount === 0) {
+			return Infinity;
+		}
+
+		let num: number = n;
+		if (isNaN(num)) {
+			num = 0;
+		}
+
+		const freeObj = this.state.elements.length - num;
+		return (this.state.objectCount - freeObj) / this.state.objectCount;
+	}
+	public isAboveThreshold(n: number = 0): boolean {
+		return this.utilization(n) >= this.state.increaseBreakPoint;
+	}
+
+	public get(): T | null {
+		if (!this.isValidState(this.state)) {
+			return null;
+		}
+
+		if (this.state.autoIncrease && this.isAboveThreshold(1)) {
+			this.increase(Math.ceil(this.state.objectCount * this.state.increaseFactor));
+		}
+
+		let result = this.state.elements.pop();
+
+		if (result === undefined) {
+			return null;
+		} else {
+			return result;
+		}
+	}
+	public allocate(n: number = 1): Array<T> {
+		if (!this.isValidState(this.state)) {
+			return [];
+		}
+
+		let num: number;
+		if (!this.isInteger(n) || n < 1) {
+			num = 1;
+		} else {
+			num = n;
+		}
+
+		let result: Array<T> = [];
+
+		while (this.state.autoIncrease && this.isAboveThreshold(num)) {
```

Comment:
Can you give some context/details about this loop?

Reply:
that was to account for calling for more elements than even one increase call could handle, but if this function is changed to allocateMutliple uses allocate that issue is handled